packages/select/src/components/select/suggest.tsx

```diff
@@ -104,10 +110,12 @@ export class Suggest<T> extends React.PureComponent<ISuggestProps<T>, ISuggestSt
 
     private renderQueryList = (listProps: IQueryListRendererProps<T>) => {
         const { inputProps = {}, popoverProps = {} } = this.props;
-        const { isOpen, selectedItem } = this.state;
+        const { isOpen } = this.state;
         const { handleKeyDown, handleKeyUp } = listProps;
         const { placeholder = "Search..." } = inputProps;
 
+        // Let's support both controlled and uncontrolled.
+        const selectedItem = this.props.selectedItem || this.state.selectedItem;
```

Comment:
this is not how i've implemented un/controlled behavior in the past. i prefer to use `this.state` as the single point of truth and to put `props` changes into state using lifecycle methods.

Reply:
also please add unit tests for this behavior.